setup.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+from glob import glob
+from os import environ
+from os.path import join, dirname, realpath
+
+from setuptools import setup, find_packages
+# Monkey patching to disable version normalization, as we are using dates with
+# leading zeroes
+# https://github.com/pypa/setuptools/issues/308
+from setuptools.extern.packaging import version as v
+
+v.Version = v.LegacyVersion
+
+
+def get_cylc_version():
+    """Get Cylc version."""
+    dir_path = dirname(realpath(__file__))
+    module_path = join(dir_path, "lib")
+    sys.path.append(module_path)
+    # TODO: Cylc __init__.py changes PYTHONPATH, breaking setup.py
```

Comment:
I think we should change `__init__.py` so the module can be safely imported, was tempted to do this in the Py3 PR. We could add a `__version__` attr.

Reply:
Oh, if you have any suggestions like this, I'm keen to hear and try to help. I suspect these improvements would make this one simpler, and perhaps contribute to #2991